packages/amplication-server/src/core/entity/entity.service.ts

```diff
@@ -607,7 +576,88 @@ export class EntityService {
           connect: {
             id: args.data.entity.connect.id
           }
+        }
+      }
+    });
+
+    return this.cloneVersionData(firstEntityVersion.id, newEntityVersion.id);
+  }
+
+  private async cloneVersionData(
+    sourceVersionId: string,
+    targetVersionId: string
+  ): Promise<EntityVersion> {
+    const sourceVersion = await this.prisma.entityVersion.findOne({
+      where: {
+        id: sourceVersionId
+      },
+      include: {
+        fields: true,
+        permissions: {
+          include: {
+            permissionRoles: true,
+            permissionFields: {
+              include: {
+                permissionRoles: true,
+                field: true
+              }
+            }
+          }
+        }
+      }
+    });
+
+    if (!sourceVersion) {
+      throw new Error(`Can't find source (Entity Version ${sourceVersionId})`);
+    }
+
+    //Find the target version and clear any existing fields and permissions (used when discarding changes and rolling back to previous version)
+    let targetVersion = await this.prisma.entityVersion.update({
+      where: {
+        id: targetVersionId
+      },
+      data: {
+        fields: {
+          deleteMany: {
+            entityVersionId: targetVersionId
+          }
+        },
+        permissions: {
+          deleteMany: {
+            entityVersionId: targetVersionId
+          }
+        }
+      }
+    });
+
+    if (!targetVersion) {
+      throw new Error(`Can't find target (Entity Version ${sourceVersionId})`);
+    }
+
+    // Duplicate the fields of the source version, omitting entityVersionId and
+    // id properties.
+    const duplicatedFields = sourceVersion.fields.map(field =>
+      omit(field, ['entityVersionId', 'id'])
+    );
+
+    //update the target version with its fields, and the its parent entity
+    targetVersion = await this.prisma.entityVersion.update({
+      where: {
+        id: targetVersionId
+      },
+      data: {
+        entity: {
+          update: {
+            name: sourceVersion.name,
+            displayName: sourceVersion.displayName,
+            pluralDisplayName: sourceVersion.pluralDisplayName,
+            description: sourceVersion.description
+          }
         },
+        name: sourceVersion.name,
+        displayName: sourceVersion.displayName,
+        pluralDisplayName: sourceVersion.pluralDisplayName,
+        description: sourceVersion.description,
```

Comment:
Since these fields are identical to the once you are passing to entity update maybe better save them to a variable and use it twice.